SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -1672,8 +1672,8 @@ public IActionResult PrivateBeta_Post(PrivateBetaViewModel viewModel)
         {
             var propertyData = await propertyDataStore.LoadPropertyDataAsync(reference);
             
-            // If entryPoint is set then the user is editing their answers, so we need to take a copy of the current answers
-            if (entryPoint is not null && propertyData.UneditedData is null)
+            // If entryPoint is set then the user is editing their answers (and if HasSeenRecommendations then they have already generated recommendations that may now need to change), so we need to take a copy of the current answers
+            if ((entryPoint is not null || propertyData.HasSeenRecommendations) && propertyData.UneditedData is null)
```

Comment:
I can't think of any issues this will cause but just to flag it's possible I'm forgetting something